### PR TITLE
Ava 168

### PR DESCRIPTION
**Pull request by kroos47**
- [x] fix on random sampling

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -184,9 +184,10 @@ pub async fn do_main() -> Result<()> {
 						log::error!("chunk size less than 3");
 					}
 					let commitment = header.extrinsics_root.commitment.clone();
+					let index_tuple = header.app_data_lookup.index.clone();
+					let kate_cells = rpc::generate_random_cells(max_rows, max_cols, num);
```
Comment (sh3ll3x3c) [inline src/main.rs:188]: great change!

File: src/rpc.rs [hunk 2]
```diff
@@ -333,30 +312,30 @@ pub fn generate_app_specific_cells(
...
 			row: rows as u16,
 			col: cols as u16,
 			..Default::default()
 		});
-	}
+	});
 	buf
 }
 
-pub fn get_id_specific_size(num: Block) -> HashMap<u32, u32> {
```
Comment (aterentic-ethernal) [inline src/rpc.rs:357]: Maybe to rename num to header or something like that. I think that `num` has confused me at some point somewhere in a codebase :D

Comment (kroos47) [reply src/rpc.rs:357]: yeh me too, just went with the flow. Will change

File: src/main.rs [hunk 3]
```diff
@@ -212,31 +213,54 @@ pub async fn do_main() -> Result<()> {
...
+									for j in 0..max_cols {
+										cells.push(types::Cell {
+											block: num,
+											row: i,
+											col: j,
+											..Default::default()
+										});
+									}
+								}
+								cells
```
Comment (aterentic-ethernal) [inline src/main.rs:231]: Maybe shorter version (not sure which one is better):

`(0..cols).flat_map(move |col| (0..rows).map(move |row| types::Cell { block: num, row, col, ..Default::default() });` 

Also we could move this to typed mod?

**Review by aterentic-ethernal — APPROVED** [review]

File: src/main.rs (hunk 3 above)
Comment (kroos47) [reply src/main.rs:231]: yeh thanks! I will try it out

**Comment by kroos47** [comment]
There is this case in `-1` that the number of cells that can be queried must be set or 50% of the total number of cells so that it can be reconstructed later.

**Pull request merged**